woocommerce/woocommerce
# [Payments NOX] Integrate incentives logic into frontend

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52453 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Run `pnpm i && pnpm build`
1. Pull the latest `trunk` on your local server installation and the latest `develop` on your local WooPayments.
2. Make sure there are no activated payment extensions, including WooPayments. (You can disable it in `WP Admin > Plugins`)
3. Checkout this PR's branch. 
4. Apply [this patch](https://gist.githubusercontent.com/vladolaru/9f9c8b18304d507a8b7e95e24962128e/raw/277a0effe4c14d4d5d8be6e94c17d58bcc8dc153/wc-96-wcpay-action-incentive.patch) to meet the Action incentive eligibility criteria.
5. Reload the Payment Settings page.
6. Verify that you can see the Incentive Banner, and that any providers which have an active incentive are highlighted in the interface matching the designs. (see screenshots). The text should match the expected text for an action incentive.
7.  Reset the previous patch that was applied.
8. Enable a different payment gateway (e.g. PayPal).
9. Apply [this patch](https://gist.github.com/dmallory42/0bfbbbea358e26a54224b535df21376a) to meet the Switch incentive eligibility criteria
10. Reload the page, you should see the modal for the switch incentive. You should also see the banner, and the row highlighting. Verify this matches the designs.

### Dismissing an Incentive

1. Apply [this patch](https://gist.githubusercontent.com/vladolaru/9f9c8b18304d507a8b7e95e24962128e/raw/277a0effe4c14d4d5d8be6e94c17d58bcc8dc153/wc-96-wcpay-action-incentive.patch) to meet the Action incentive eligibility criteria.
2. Reload the Payment Settings page.
3. Verify that you can see the Incentive Banner.
4. Click `No thanks` on the Incentive Banner.
5. You should see a request being made to the `/dismiss` endpoint. Verify it returns successfully.
6. The incentive should disappear. Reload the page and the incentive should no longer show. 
7. Reset the previous patch that was applied.

### Accepting an incentive

1. Follow the steps in the troubleshooting section to clear the state of your store so that you can see the extension again.
2. Apply [this patch](https://gist.githubusercontent.com/vladolaru/9f9c8b18304d507a8b7e95e24962128e/raw/277a0effe4c14d4d5d8be6e94c17d58bcc8dc153/wc-96-wcpay-action-incentive.patch) to meet the Action incentive eligibility criteria.
3. Verify you can see the incentive banner. Click the button to accept the incentive.
4. Onboard the account. You should see the incentive details on the `Payments > Overview` page (under account status).

### Troubleshooting

If you don't see the incentives, there are various things you can try to make sure your store is in the right state:

- Make sure you have Marketplace suggestions enabled in `WooCommerce > Settings > WooCommerce.com`
- Delete the `woocommerce_task_list_hidden` option.
- Use the WC Beta Tester plugin to reset WC onboarding and set the timestamp to now. (see screenshot)
- Delete any `wp_option` rows with `incentive` in them (you can query `where option_name LIKE '%incentive%'`

<img width="1297" alt="Screenshot 2024-12-09 at 18 36 23" src="https://github.com/user-attachments/assets/77047d61-bc19-4865-98ba-35a1ca8caa90">



### Screenshots

Incentive banner and row highlighting:

<img width="2194" alt="Screenshot 2024-12-05 at 18 41 05" src="https://github.com/user-attachments/assets/9d5cf25a-737b-4589-b4e4-75660e47279b">

### Designs

xO8YdpKfaVoP2d8NvweZ5I-fi-3106_71471

### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
